pychunkedgraph/tests/test_uncategorized.py

```diff
@@ -1412,7 +1412,10 @@ def test_cross_edges(self, gen_graph):
 
         for layer in n_cross_edges_layer.keys():
             cgraph.logger.debug("LAYER %d" % layer)
-            assert len(np.unique(n_cross_edges_layer[layer])) == 1
+            if layer == 5:
```

Comment:
looks good overall but why hard code layer `5` in the unit test?

Reply:
It's a little complex, so I'll explain here. The unit test creates a ChunkedGraph with 5 layers and 1 root. It then calculates for the subgraph of the root at each layer, how many cross chunk edges that become relevant at each layer there are, the result looks like this:

```
2 2 8
2 3 2
2 4 0
2 5 4

3 3 2
3 4 0
3 5 4

4 4 0
4 5 2

5 5 4
```

The first column is the layer of the subgraph, the second column is the layer that the cross chunk edges of the subgraph become relevant, and the third column is the number of cross chunk edges for this configuration. For example for the layer 2 subgraph, 8 cross chunk edges are relevant at layer 2 and 2 are layer at level 3. At the layer 4 subgraph, 2 are relevant at layer 5 (the line "4 5 2"). Before the `get_subgraph_nodes` bug fix included in this PR, this line was "4 5 4", because 2 level 2 ids were included in the layer 4 subgraph which shouldn't have been included, and this resulted in 2 more cross chunk edges being counted. That is why I added this check: now layer 4 has a different amount of cross chunk edges that become relevant at layer 5 than the other layers -- the number of cross chunk edges that become relevant at layer 5 now has 2 unique values (as opposed to other layers). To summarize, I added this check because the test's view of this local ChunkedGraph has now changed, which makes the old assertion incorrect in the case of layer 5.

Reply:
okay I didn't realize that was a 5 layer chunkedgraph, I think it would be better to use a variable though, same one used with parameter `n_layers` passed to `gen_graph()`. If the test changes this can be missed easily. Or even better `cg.meta.layer_count` if possible.

Reply:
My mistake -- it's actually a 6 layer ChunkedGraph, the 5th layer is the highest requested subgraph. I pushed a commit where I updated the check using the `cg.n_layers` param, and included a short comment.